File: src/main/java/com/bajdcc/LALR1/syntax/automata/npa/NPAStatusData.java
package com.bajdcc.LALR1.syntax.automata.npa;

/**
 * 非确定性下推自动机状态数据
 *
 * @author bajdcc
 */
public class NPAStatusData {
	/**
	 * 标签
	 */
	public String label = "";

	/**
	 * 状态所属规则的索引
	 */
	public int iRuleItem = -1;
}

File: src/main/kotlin/com/bajdcc/LALR1/syntax/automata/npa/NPAStatusData.kt
package com.bajdcc.LALR1.syntax.automata.npa

/**
 * 非确定性下推自动机状态数据
 * [label] 标签
 * [rule] 状态所属规则的索引
 * @author bajdcc
 */
data class NPAStatusData(var label: String = "", var rule: Int = -1)